src/Orleans/Streams/PubSub/ImplicitStreamSubscriberTable.cs

```diff
@@ -133,16 +133,19 @@ private Guid MakeSubscriptionGuid(GrainId grainId, StreamId streamId)
         /// <returns></returns>
         private Guid MakeSubscriptionGuid(int grainIdTypeCode, StreamId streamId)
         {
+            JenkinsHash jenkinsHash = JenkinsHash.Factory.GetHashGenerator();
+
             // next 2 shorts ing guid are from namespace hash
-            int namespaceHash = streamId.Namespace.GetHashCode();
+            uint namespaceHash = jenkinsHash.ComputeHash(streamId.Namespace);
             byte[] namespaceHashByes = BitConverter.GetBytes(namespaceHash);
             short s1 = BitConverter.ToInt16(namespaceHashByes, 0);
             short s2 = BitConverter.ToInt16(namespaceHashByes, 2);
 
             // Tailing 8 bytes of the guid are from the hash of the streamId Guid and a hash of the full streamId.
 
             // get streamId guid hash code
-            int streamIdGuidHash = streamId.Guid.GetHashCode();
+            uint streamIdGuidHash = jenkinsHash.ComputeHash(streamId.Guid.ToByteArray());
+
             // get full streamId hash code
             int streamIdHash = streamId.GetHashCode();
```

Comment:
actually, instead of using  streamId.GetUniformHashCode here I would take provider name and compute Jenkins on it. that way you have namespace, Guid and provider name all explicitly captured and hashed directly here.